src/_response.ts

```diff
@@ -0,0 +1,62 @@
+import {
+  CHALK_DATE_TYPES,
+  ChalkOnlineQueryRawData,
+  ChalkOnlineQueryRawResponse,
+} from "./_http";
+import { ChalkOnlineQueryResponse } from "./_interface";
+import { ChalkClientConfig, ChalkScalar } from "./_types";
+import { fromEntries } from "./_utils";
+import { mapRawResponseMeta } from "./_meta";
+
+type FeatureEntry<
+  TFeatureMap = Record<string, ChalkScalar>,
+  TOutput extends keyof TFeatureMap = keyof TFeatureMap
+> = ChalkOnlineQueryResponse<TFeatureMap, TOutput>["data"][any];
+
+const processDataValue = (
+  d: ChalkOnlineQueryRawData,
+  { timestampFormat }: Pick<ChalkClientConfig, "timestampFormat">
+): ChalkScalar => {
+  if (
+    timestampFormat === "EPOCH_MILLIS" &&
+    CHALK_DATE_TYPES.has(d.meta?.primitive_type)
+  ) {
+    return new Date(d.value).getTime();
+  }
+
+  return d.value;
+};
+
+export const parseOnlineQueryResponse = <
+  TFeatureMap = Record<string, ChalkScalar>,
+  TOutput extends keyof TFeatureMap = keyof TFeatureMap
+>(
+  rawResult: ChalkOnlineQueryRawResponse,
+  config: Pick<ChalkClientConfig, "timestampFormat">
+): ChalkOnlineQueryResponse<TFeatureMap, TOutput> => {
+  return {
+    data: fromEntries(
+      rawResult.data.map((d): [string, FeatureEntry] => {
+        return [
+          d.field,
+          {
+            value: processDataValue(d, config),
```

Comment:
I wonder if it's worth moving the `timestampFormat` check to the outer level of `parseOnlineQueryResponse` rather than running it for every feature value - I'm sure the efficiency gain is not much and the code style might suffer but something in me feels that it's wasteful to check it for every value when we only really need to check it once and then we can shortcut a bunch of function calls. But hyperefficiency in the typescript client might not be our biggest concern at the moment, so I'd understand if you just want to keep it as is.